partiql-value/src/graph.rs

```diff
@@ -59,36 +58,45 @@ impl Ord for Graph {
     }
 }
 
-#[derive(Debug, Clone, Hash, PartialEq, Eq)]
+#[derive(Debug, Copy, Clone, Hash, PartialEq, Eq)]
 pub struct GNodeId(pub usize);
 
-#[derive(Debug, Clone, Hash, PartialEq, Eq)]
+#[derive(Debug, Copy, Clone, Hash, PartialEq, Eq)]
 pub struct GEdgeId(pub usize);
 
-#[derive(Clone, Hash, PartialEq, Eq)]
+#[derive(Debug, Copy, Clone, Hash, PartialEq, Eq)]
 pub struct GLabelId(pub Spur);
 
 #[derive(Clone, PartialEq, Eq)]
 pub struct GLabels(pub HashSet<GLabelId>);
 
 #[derive(Clone, PartialEq, Eq)]
-pub struct GElem(pub Option<Value>, pub GLabels);
+pub struct GElem {
+    pub value: Option<Value>,
+    pub labels: GLabels,
+}
+
+impl GElem {
+    pub fn new(value: Option<Value>, labels: GLabels) -> Self {
+        GElem { value, labels }
+    }
+}
 
 pub struct SimpleGraph {
-    directed: petgraph::stable_graph::StableGraph<GNodeId, GEdgeId, Directed>,
-    undirected: petgraph::stable_graph::StableGraph<GNodeId, GEdgeId, Undirected>,
```

Comment:
out of curiosity, what's the rationale to use our own representation rather than use petgraph?

Reply:
The complexity of dealing with two different graphs and trying to implement the GPML algorithm on-top of petgraph made it hard to follow. For something intended to be faster/better/etc. than a 'SimpleGraph', petgraph may be better, but it obfuscated the implementation here.

Reply:
I add that that it is one less dependency which is also a plus; obviously considering that it seems, it is making the implementation harder, based on @jpschorr's note.